### PR TITLE
Fixes #6280 : Added right associativity to 'pow' function

### LINKED ISSUE #6280: Builtin math: `^` operator is left-associative
The result of `math '3^0.5^2'` is currently:
```
~> math '3^0.5^2'
3
```
That mean, `^` operator is left-associative.
I think this is a bug because exponentiation operators in many languages are right-associative.

version:
```
~> fish --version
fish, version 3.0.2-1757-g47c0b5f9
```

### PR DESCRIPTION
**Pull request by asprazz**
## Description
The pow function considered to be left associative

Talk about your changes here.
After comparing with upstream I have added a constant variable which determines the associativity at run time

Fixes issue #6280 

## TODOs:
I am first time contributor to this awesome project so please guide me through this..

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [x] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: src/tinyexpr.cpp [hunk 1]
```diff
@@ -22,6 +22,15 @@
...
  */
 
+/* COMPILE TIME OPTIONS */
+
+/*
+ * Exponentiation associativity:
+ * For a^b^c = a^(b^c) do nothing
+ * For a^b^c = (a^b)^c comment the next line.
+ */
+#define TE_POW_FROM_RIGHT
```
Comment (faho) [inline src/tinyexpr.cpp]: Please just remove the option entirely. This isn't a library anymore.

Comment (asprazz) [reply src/tinyexpr.cpp]: Means just want only one version of pow? and I will squash changes..

**Comment by faho** [comment]
It'd be nice to have some tests for this - we don't want to accidentally go back.

See tests/checks/math.fish.

File: src/tinyexpr.cpp (hunk 1 above)
Comment (asprazz) [reply src/tinyexpr.cpp]: Done @faho :) Please check if that was expected sorry I hard pushed two times and amended changes with last commit

**Comment by floam** [comment]
Nice job!

**Pull request merged**